### PR TITLE
New pagebreak plugin

### PR DESCRIPTION
**Pull request by n9iels**
I made this addition as requested by @mbabker in #7750 
### This PR conatin
- JHtml bootstrap sliders and tabs
- JLayout implementation for the navigation
- Code style / code cleaning
- Rename the old Mootools tabs to `Mootools tabs` and `Mootools slider`. The new tabs made with Bootstrap have the name `Tabs` and `Sliders`
### Test instructions
1. Apply the patch, and confirm that all parameters are working like they used to do. (Please notice the "Presentation Style" parameter values `Tabs` and `Sliders` are the new Bootstrap Tabs).
### Notes

I'm not sure if this plugin is allready correct and does what it has to do, also I'm not a pro developer. I simply made addition to the old plugin. So please comment on my code, give me hints and tips and tell me if I'm on the good why or not :blush:

### CODE REVIEW

**Comment by mbabker** [comment]
At a quick glance it looks like a good start.

So there's two options going forward.  Option 1 like I originally thought of is a totally new plugin with just the Bootstrap styling options and the pages option.  Option 2 is to add two new styles to the existing plugin, 'Bootstrap Sliders' and 'Bootstrap Tabs', and everything can be supported in one plugin.  Either way, the existing sliders and tabs style options need to be preserved in a B/C manner because changing from the JHtml helpers producing the MooTools based content for those to the Bootstrap versions will cause layout breaks in sites.  Option 2 is probably a cleaner choice.

**Comment by n9iels** [comment]
I also think the second option is the best. I will add them to these PR

**Comment by n9iels** [comment]
I re-add the Mootools tabs & slider option, and make new select option for the XML file. Also fix a little bug in the Table of Content custum heading.

**Comment by N6REJ** [comment]
why can't there be dual class's?  for example for bootstrap 2 & 3 
compatibility you could have span2 col-md-2
With our current frontend being BS2 and HOPEFULLY soon bs3 or bs4 
wouldn't it make sense to stick with BS?  Everyone has had plenty of 
warning to get away from mootools
Bear

**Comment by mbabker** [comment]
Splitting things into layouts makes it possible to override the HTML without hacking the plugin.  So it's less of an issue to have BS2 and BS3+ syntax in this location.  There are some places with non-overridable markup I'd agree with you on that though, and one was merged recently (the debug plugin).

**Comment by n9iels** [comment]
offtopic of this plugn:
@N6REJ I think in the feature using one framework in Joomla! issn't the best idea. In my opinium it is better to write a own Joomla! CSS framework, that can be loaded next to another framework like Bootstrap or Zurb. With someting like that, you can use every framework you want, without conflicts and backward compability problems.

**Comment by dgt41** [comment]
I have tested this item :white_check_mark: successfully on a4f1e190086dd2788f3ac98fb7242333c161c02e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7760">issues.joomla.org/joomla-cms/7760</a>.</sub>

**Comment by schnuti** [comment]
:white_check_mark:  I have tested this item successfully

I propose a change of the option names. The JLayouts my be overridden using other frameworks than Bootstrap. ie the new options should not refer to it. I have no nice name at hand though. 
'Tabs new style' and 'Sliders new style' ?

There is still a problem with the old sliders. They do not work. This is not a stop for this PR! as it presents a working alternative.

**Comment by waader** [comment]
I have tested this item :white_check_mark: successfully on a4f1e190086dd2788f3ac98fb7242333c161c02e<br /><br />Thanks @niels! I have tested with current Chrome, FF, IE and IE8.

@Schnuti: what operating system and browser do you use?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7760">issues.joomla.org/joomla-cms/7760</a>.</sub>

**Comment by schnuti** [comment]
@waader: I tested localy XAMPP, Win 10 and Firefox

**Comment by waader** [comment]
As I totally refuse Win10. Do you experience the same probleme on Win7/8 and FF respectively do you have the possibility to test this combination?

**Comment by schnuti** [comment]
@waader Do you mean test the old sliders?

I have to withdraw my statement that it's not working. I hope I was right about it before this PR.
I now also tested wih the new MS Edge browser.
I presumably didn't click on the slider names now. It is confusing that slide-1 (page 2) is opend when you open the article page. ie the first page should be very short when using sliders.
It's the same with tabs. I never tested tabs and sliders before #8279 

To me it would be better to include page 1 in a first opened slider/tag but that's another issue.
--------------------------------------------------------------------------------------------------------------------------
Win 10 is not much worse than Win7. I've removed most of the new 'features' and for sure am not using any spying apps. My netbook with win7 gave up some time ago. I still didn't decide how to replace it.

**Comment by n9iels** [comment]
The sliders are working correct for me on all browsers (I mean, they slide, hide content and make it visible when I click on it). But it is true that a bootstrap slider is not open by default, and for the mootools slider (slide old style) the first slider is always open by default.

Not sure if that is something in this PR or the old slider API. I have to check that later.

**Comment by schnuti** [comment]
@n9iels
the default opened sliders/tabs are in the old api and should probably not be changed. Who knows who depend on it? 
But give a thougt to the naming of the options. Not Bootstrap?

**Comment by n9iels** [comment]
I think I solve the merge conflict now.

@schnuti Good point about the name, it may be a good option.
If the sliders should open by default is something what can be discussed in a PR or Issue. If it is a simple change and you can do it, make a PR and we can discuss over there if it is a good idea or not.

**Comment by schnuti** [comment]
@n9iels
You misunderstood. Please do not change the closed tabs/sliders for the new options. I meant that maybe it's better to keep the old behavior for the old options.

**Comment by n9iels** [comment]
Ah, that way, I won't :wink:

File: administrator/language/en-GB/en-GB.plg_content_pagebreak.ini [hunk 1]
```diff
@@ -18,9 +18,11 @@ PLG_CONTENT_PAGEBREAK_SITE_ARTICLEINDEX_DESC="Show or hide Article Index Heading
 PLG_CONTENT_PAGEBREAK_SITE_ARTICLEINDEX_LABEL="Article Index Heading"
 PLG_CONTENT_PAGEBREAK_SITE_TITLE_DESC="Title and heading attributes from Plugin added to Site Title tag."
 PLG_CONTENT_PAGEBREAK_SITE_TITLE_LABEL="Show Site Title"
-PLG_CONTENT_PAGEBREAK_SLIDERS="Sliders"
 PLG_CONTENT_PAGEBREAK_STYLE_DESC="Choose whether to layout the article with separate pages, tabs or sliders."
 PLG_CONTENT_PAGEBREAK_STYLE_LABEL="Presentation Style"
+PLG_CONTENT_PAGEBREAK_BOOTSTRAPSLIDERS="Bootstrap Sliders"
```
Comment (dgt41) [inline administrator/language/en-GB/en-GB.plg_content_pagebreak.ini]: @n9iels you need to alpha order the strings here

**Comment by infograf768** [comment]
Please re-order this string:
```
PLG_CONTENT_PAGEBREAK_SITE_TITLE_LABEL="Show Site Title"
-PLG_CONTENT_PAGEBREAK_SLIDERS="Sliders"
 PLG_CONTENT_PAGEBREAK_STYLE_DESC="Choose whether to layout the article with separate pages, tabs or sliders."
 PLG_CONTENT_PAGEBREAK_STYLE_LABEL="Presentation Style"
+PLG_CONTENT_PAGEBREAK_SLIDERS="Sliders"
```

**Comment by n9iels** [comment]
Sorry for the double commit.
@infograf768 I alpha sort the whole language file now

File: plugins/content/pagebreak/pagebreak.php [hunk 2]
```diff
@@ -340,41 +326,32 @@ protected function _createToc(&$row, &$matches, &$page)
...
 			$link_prev = JRoute::_(ContentHelperRoute::getArticleRoute($row->slug, $row->catid, $row->language) . '&showall=&limitstart=' . ($page_prev));
-
-			// << Prev
-			$prev = '<a href="' . $link_prev . '">' . JText::_('JGLOBAL_LT') . JText::_('JGLOBAL_LT') . $pnSpace . JText::_('JPREV') . '</a>';
 		}
 		else
 		{
-			$prev = JText::_('JPREV');
+			$link_prev  = null;
+			$prev 		= JText::_('JPREV');
```
Comment (zero-24) [inline plugins/content/pagebreak/pagebreak.php]: can wr fix the indend here ;)

Comment (zero-24) [inline plugins/content/pagebreak/pagebreak.php]: Fixes here: https://github.com/n9iels/joomla-cms/pull/3

**Comment by n9iels** [comment]
I change the name of the old sliders and tabs to "Mootools slider" and "Mootools Tabs". The value of that option stays the same, so there is no BC problem.

Also, I add two new JLayout files for the New tabs and New slider. So they can be overridden.

**Comment by schnuti** [comment]
:white_check_mark: I tested to override navigation and toc succsessfully.

BUG: If no title is set to the pagebreak line 111 fails. Paremeter = Page! ie not Sliders/Tabs

I'm not sure if a default should be added to the page title. Probably yes.

This works for me

    $attrs = JUtility::parseAttributes($matches[$page - 1][1]);
    if (isset($attrs['title']))
    {
       $row->page_title = $attrs['title'];
    }
       else
    {
       $row->page_title = JText::sprintf('PLG_CONTENT_PAGEBREAK_PAGE_NUM', $page + 1);
    }

**Comment by schnuti** [comment]
@n9iels
I think you have missunderstood JLayout a bit. What you added is "layouts" in the tmpl folder. Those can be overridden. JLayouts are reusable and for Joomla stored in path layouts/

It is correct that you now can override to eg choose another JLayout.
JHtml::_('bootstrap. - is partly (for tabs) using JLayout from layouts/libraries/cms/html/bootstrap.
A new project could be to add sliders to this path, if not allready in work. 

:heavy_plus_sign: What you added is though very very usefull!

It's hard to get all those levels, but as I understand, it's the way to a less Bootstrap 2 lock in.

**Comment by n9iels** [comment]
@schnuti You mean I should add the layout folder off the plugin to the root/layout folder? Now you note that it make sense indeed. Overriding these layout now via a template is quite difficult. Is that what you mean?

And that is true, and exactly what I want :). The idea is we can easy override these JLayout and use Bootstrap 3 code instead of JHtml Bootstrap.

**Comment by schnuti** [comment]
@n9iels No, I think you have gone as far as possible right now. The rest has to find a common Joomla way. You can now override the JLayout files for the Bootstrap tabs. Those overrides are system relevant. You can also add other javascripts to your template to override the 'common' Bootstrap js. 
ie your tabs would probably work with Bootstrap 3 and 4 if you add this code to your template.

But I think the navigation could be converted to a JLayout. ie the navigation.php calling a JLayout. If someone wants a special output for the article paging they can still override navigation.php otherwise a common JLayout is used. I don't know, if there is one allready. Will have a look. 

I found a few Bootstrap 2 classes in tabelofcontent.php. It could be overridden or use two new parameters.
with default "pull-right" for the div wrapper 
and "nav nav-tabs nav-stacked" for the ul class.
That is my proposal.

**Comment by schnuti** [comment]
@n9iels :smirk: You promised not to change the closed tabs. You did!
newtabs.php line 13

    $t[] = JHtml::_('bootstrap.startTabSet', 'pageBreak', array('active' => 'article-1'));

Please remove article-1

    $t[] = JHtml::_('bootstrap.startTabSet', 'pageBreak', array('active' => ''));

The sliders are closed!

**Comment by n9iels** [comment]
Woops. But, for the old tabs the first tab is always open... Maybe for consistency it is better to do the same for the Bootstrap Tabs?

I moved the navigation to JLayout so it can be overridden. For that two bootstrap 2 classes I don't know a solution at the moment. Convert tableofcontent.php to a JLayout is not a good idea, because it is a complex file with parameters etc.

**Comment by schnuti** [comment]
Open tabs and sliders would be a no go for me and probably for most users. Two pages directly shown when you open an article? Impossible. The new options can only do it better than the old. 

The JLayouts are only for common reusable output. Not for things like tableofcontent.php.
I don't understand why you changed navigation.php to JLayout. What advantage? You only changed path for the override.
It has to be added to layouts/joomla/somewhereto give an advantage.

**Comment by n9iels** [comment]
The advantage is that the navigation can be overridden now via a template override: `html/layouts/plugins/content/pagebreak/navigation.php`. That is in my opinion the best way, so users can easily change it if they like to.

**Comment by conconnl** [comment]
@n9iels is this PR ready for testing?
Can you write a full test scenario with Before patch and after patch?
Then we can let people test it on the next PBF (probably August 20, Eindhoven)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/7760">issues.joomla.org/joomla-cms/7760</a>.</sub>

**Comment by n9iels** [comment]
No this PR isn't ready for testing at all. I don't have time at the moment to make it ready, also the code is not good. So I close this PR, and maybe I make a new one when I have time for it.

If someone else like to do it, no problem at all :+1:
